app/editor/src/features/admin/sources/DataSourceForm.tsx

```diff
@@ -58,6 +59,10 @@ export const DataSourceForm: React.FC<IDataSourceProps> = (props) => {
     } catch {}
   };
 
+  const hasErrors = (errors: any, props: string[]) => {
```

Comment:
I don't love this solution, but it's all I could think of today.  Essentially just checks if an error property is positive.